sirepo/pkcli/db.py

```diff
@@ -34,3 +34,86 @@ def _inc(m):
                 break
             with open(str(fn), 'w') as f:
                 f.write(t)
+
+
+def seed_supervisor_state(db_dir, sbatch_poll_secs):
+    from pykern import pkio
+    from pykern.pkcollections import PKDict, json_load_any
+    from pykern.pkdebug import pkdp
+    from sirepo import job
+    from sirepo import simulation_db
+    from sirepo import util
+    from sirepo import sim_data
+    import sirepo.template
+    import os
+
+    db_dir = pkio.py_path(db_dir)
+    pkio.mkdir_parent(db_dir)
+
+    _NEXT_REQUEST_SECONDS = PKDict({
+        job.PARALLEL: 2,
+        job.SBATCH: int(sbatch_poll_secs),
+        job.SEQUENTIAL: 1,
+    })
+
+    def _add_parallel_status(in_json, sim_data, run_dir, data):
+        t = sirepo.template.import_module(data.simulationType)
+        data.parallelStatus = PKDict(
+           t.background_percent_complete(
+               sim_data.parse_model(in_json),
+               run_dir,
+               False,
+           )
+        )
+
+    def _create_supervisor_state_file(uid, run_dir):
+        i = json_load_any(
+            run_dir.join(
+                sirepo.template.template_common.INPUT_BASE_NAME +
+                simulation_db.JSON_SUFFIX
+            )
+        )
+        c = sim_data.get_class(i.simulationType)
+        p = c.is_parallel(i)
```

Comment:
I'm confused between `p`, `j`, and the use of `d.isParallel` below. Should keep consistency. Perhaps you would assign d.isParallel first, and then pkupdate `d` to include things that depend on isParallel, [like is done in job_api.py](https://github.com/radiasoft/sirepo/blob/5f9f5783209679d663b24bac364ea76641b90e59/sirepo/job_api.py#L194)

Reply:
I've tried to make it clearer. Let me know what you think.

Reply:
Clearer now.